tensorflow/contrib/seastar/seastar_server_lib.cc

```diff
@@ -0,0 +1,260 @@
+    ParseGrpcServerDef(server_def);
+    LoadEndpointMapForFile();
+  }
+
+  std::string GetSeastarIpPort(const std::string& job_name, int task_index) {
+    const auto it_job = grpc_cluster_spec_.find(job_name);
+    if (it_job == grpc_cluster_spec_.end()) {
+      LOG(FATAL) << "Job name: " << job_name
+                 << " does not exist in cluster spec.";
+    }
+    const std::map<int, std::string>& task_map = it_job->second;
+
+    const auto it_task = task_map.find(task_index);
+    if (it_task == task_map.end()) {
+      LOG(FATAL) << "Job name: " << job_name << ", task index: " << task_index
+                 << " does not exist in cluster spec.";
+    }
+    const std::string& grpc_ip_port = it_task->second;
+
+    const auto it_seastar = endpoint_grpc2seastar_.find(grpc_ip_port);
+    if (it_seastar == endpoint_grpc2seastar_.end()) {
+      LOG(FATAL) << "Seastar ip and port not found for job name: " << job_name
+                 << "task index: " << task_index << ".";
+    }
+
+    return it_seastar->second;
+  }
+
+  int GetLocalSeastarPort() {
+    const auto it = endpoint_grpc2seastar_.find(local_grpc_ip_port_);
+    if (it == endpoint_grpc2seastar_.end()) {
+      LOG(FATAL) << "Seastar ip and port not found for job name: " << job_name_
+                 << "task index: " << task_index_ << ".";
+    }
+    const std::string& local_seastar_ip_port = it->second;
+    std::vector<std::string> vec = str_util::Split(local_seastar_ip_port, ":");
+    CHECK_EQ(vec.size(), 2);
+
+    int local_seastar_port = -1;
+    strings::safe_strto32(vec[1], &local_seastar_port);
+    CHECK_GT(local_seastar_port, 0);
+
+    return local_seastar_port;
+  }
+
+  std::string get_job_name() const {
+    return job_name_;
+  }
+
+private:
+  void ParseGrpcServerDef(const ServerDef& server_def) {
+    job_name_ = server_def.job_name();
+    task_index_ = server_def.task_index();
+
+    for (const auto& job : server_def.cluster().job()) {
+      std::map<int, std::string>& task_map = grpc_cluster_spec_[job.name()];
+      for (const auto& task : job.tasks()) {
+        task_map[task.first] = task.second;
+        if (job.name() == job_name_ && task.first == task_index_) {
+          local_grpc_ip_port_ = task.second;
+        }
+      }
+    }
+
+    if (local_grpc_ip_port_.empty()) {
+      LOG(FATAL) << "Job name: " << job_name_ << ", task index: " << task_index_
+                 << " not found in cluter spec.";
+    }
+  }
+
+  void LoadEndpointMapForFile() {
+    std::ifstream fin(kEndpointMapFile, std::ios::in);
+    if (!fin.good()) {
+      LOG(FATAL) << "Load endpoint map file failed.";
+    }
+
+    string str;
+    while (getline(fin, str)) {
+      std::vector<std::string> vec = str_util::Split(str, '=');
+      CHECK_EQ(vec.size(), 2);
+      endpoint_grpc2seastar_[vec[0]] = vec[1];
+    }
+  }
+
+private:
+  std::map<std::string, std::string> endpoint_grpc2seastar_;
+  std::map<std::string, std::map<int, std::string> > grpc_cluster_spec_;
```

Comment:
Do you need ordering on the "key" of the std::maps? If not, I would recommend using std::unordered_map

Reply:
> Do you need ordering on the "key" of the std::maps? If not, I would recommend using std::unordered_map

fixed.